feat: add `math/base/special/expf`

vivekmaurya001 opened the pull request:
Progresses #649 

## Description

- adds math/base/special/expf, which would be the single-precision equivalent for [math/base/special/exp](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/exp).

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
added expf implementation

vivekmaurya001 committed:
test file updated

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

gunjjoshi commented:
@vivekmaurya001 Thanks for working on this, CI tests must pass in order to move ahead with this PR.

vivekmaurya001 committed:
updated main.js

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
added fixtures

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
scaled tolerance value

vivekmaurya001 committed:
changed test.native file

vivekmaurya001 committed:
changed the computation methods

vivekmaurya001 committed:
updated main.c

vivekmaurya001 committed:
updated main.c

vivekmaurya001 committed:
updated main.c values

vivekmaurya001 commented:
done changes ,  waiting for your inputs in this @gunjjoshi !

kgryte commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,233 @@
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static float polyval_p( const float x ) {
```
This function does not appear to be auto-generated using the appropriate package script.

vivekmaurya001 replied:
i have updated the coefficients in evalpoly now it is getting generated  @kgryte

vivekmaurya001 committed:
Resolved coeffiecients in evalpoly

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
feat(math): add math/base/special/expf

vivekmaurya001 committed:
feat(math): add math/base/special/expf

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
feat(math): add math/base/special/expf

vivekmaurya001 commented:
@gunjjoshi , @kgryte , Kindly give some furthur review on this for any change or anything

gunjjoshi commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

gunjjoshi commented:
@vivekmaurya001 Can you please update all the JS and C benchmarks in this PR to generate values beforehand, not in the loop?

gunjjoshi commented:
You can refer: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/ldexpf/benchmark/benchmark.js and https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/ldexpf/benchmark/c/native/benchmark.c#L92.

vivekmaurya001 committed:
Updated benchmarks

vivekmaurya001 committed:
Updated benchmarks

vivekmaurya001 commented:
@gunjjoshi , done that !

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,135 @@
+static double benchmark( void ) {
+	double elapsed;
+	float x[ 100 ];
+	float y[ 100 ];
```
@vivekmaurya001 The CI errors are arising from here.
`y` shouldn't be an array here. It must be of the type `double`.

vivekmaurya001 replied:
thanks @gunjjoshi for pointing out , im correcting it

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,72 @@
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
```
```suggestion
var randu = require( '@stdlib/random/array/uniform' );
```
The failure in running benchmarks should be resolved by this.

vivekmaurya001 replied:
ook

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.native.js:
@vivekmaurya001 You can update this file too, to use `@stdlib/random/array/uniform`.

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
error-resolve

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

gunjjoshi commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Add-C-implementations-to-base-special-math-functions

gunjjoshi commented:
@vivekmaurya001 There are some lint errors arising from the return values in JS files. Can you please look over them?

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
resolved js outputs

vivekmaurya001 committed:
Merge branch 'Add-C-implementations-to-base-special-math-functions' of https://github.com/vivekmaurya001/stdlib into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 commented:
@gunjjoshi  , resolved errors and thanks for suggested changes

kgryte commented:
@gunjjoshi Would you mind giving this another review?

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

vivekmaurya001 committed:
refactored examples and readme

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+
+# expf
+
+> Natural [exponential function][exponential-function]
```
```suggestion
> Evaluate the natural [exponential function][exponential-function] as a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
````diff
@@ -0,0 +1,230 @@
+
+```javascript
+var v = expf( 4.0 );
+// returns 54.598148345947266
````
```suggestion
// returns ~54.5982
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+// returns 54.598148345947266
+
+v = expf( -9.0 );
+// returns 0.00012340980174485594
```
```suggestion
// returns ~1.234e-4
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
````diff
@@ -0,0 +1,230 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
+var expf = require( '@stdlib/math/base/special/expf' );
+
+var x;
+var i;
+
+for ( i = 0; i < 100; i++ ) {
+    x = (randu()*100.0) - 50.0;
+    console.log( 'e^%d = %d', x, expf( x ) );
+}
````
@vivekmaurya001 We can use `@stdlib/random/array/uniform` here and initialize `x` as an array, thus pre-computing the values. For reference, see: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/lcmf/README.md?plain=1#L104

gunjjoshi replied:
This can be applied to `examples/index.js` as well.

vivekmaurya001 replied:
Done the changes !

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
````diff
@@ -0,0 +1,230 @@
+Evaluates the natural [exponential function][exponential-function] as a single-precision floating-point number.
+
+```c
+float out = stdlib_base_expf( 4.0 );
````
```suggestion
float out = stdlib_base_expf( 4.0f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
````diff
@@ -0,0 +1,230 @@
+
+```c
+float out = stdlib_base_expf( 4.0 );
+// returns 54.598148345947266
````
```suggestion
// returns ~54.5982f
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+float out = stdlib_base_expf( 4.0 );
+// returns 54.598148345947266
+
+out = stdlib_base_expf( -9.0 );
```
```suggestion
out = stdlib_base_expf( -9.0f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+// returns 54.598148345947266
+
+out = stdlib_base_expf( -9.0 );
+// returns 0.00012340980174485594
```
```suggestion
// returns ~1.234e-4f
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/math/base/special/exp10`][@stdlib/math/base/special/exp10]</span><span class="delimiter">: </span><span class="description">base 10 exponential function.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/exp2`][@stdlib/math/base/special/exp2]</span><span class="delimiter">: </span><span class="description">base 2 exponential function.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/expm1`][@stdlib/math/base/special/expm1]</span><span class="delimiter">: </span><span class="description">compute expf(x) - 1.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/ln`][@stdlib/math/base/special/ln]</span><span class="delimiter">: </span><span class="description">evaluate the natural logarithm of a double-precision floating-point number.</span>
+
```
```suggestion
```

gunjjoshi replied:
We shouldn't add anything in the `related` sections, they're populated automatically.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/README.md:
```diff
@@ -0,0 +1,230 @@
+
+<!-- <related-links> -->
+
+[@stdlib/math/base/special/exp10]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/exp10
+
+[@stdlib/math/base/special/exp2]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/exp2
+
+[@stdlib/math/base/special/expm1]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/expm1
+
+[@stdlib/math/base/special/ln]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/ln
+
```
```suggestion
```

gunjjoshi replied:
Same comment.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,85 @@
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
+var Float32Array = require( '@stdlib/array/float32' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
@vivekmaurya001 In this file and in `benchmark.native.js`, we should use `@stdlib/random/array/uniform`, instead of the current `@stdlib/random/base/randu`. For reference, see: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/fmodf/benchmark/benchmark.js

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,85 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = expf( x[ i % len ] );
+		if ( isnan( y ) ) {
```
```suggestion
		if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,85 @@
+		}
+	}
+	b.toc();
+	if ( isnan( y ) ) {
```
```suggestion
	if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,85 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = Math.exp( x[ i % len ] ); // eslint-disable-line stdlib/no-builtin-math
+		if ( isnan( y ) ) {
```
```suggestion
		if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,85 @@
+		}
+	}
+	b.toc();
+	if ( isnan( y ) ) {
```
```suggestion
	if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,67 @@
+var bench = require( '@stdlib/bench' );
+var Float32Array = require( '@stdlib/array/float32' );
+var randu = require( '@stdlib/random/base/randu' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
```
```suggestion
var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,67 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = expf( x[ i % len ] );
+		if ( isnan( y ) ) {
```
```suggestion
		if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,67 @@
+		}
+	}
+	b.toc();
+	if ( isnan( y ) ) {
```
```suggestion
	if ( isnanf( y ) ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+	return (double)now.tv_sec + (double)now.tv_usec / 1.0e6;
+}
+
+/**
+* Generates a random number on the interval [min,max).
+*
+* @param min    minimum value (inclusive)
+* @param max    maximum value (exclusive)
+* @return       random number
+*/
+static float random_uniform( const float min, const float max ) {
+	float v = (float)rand() / ( (float)RAND_MAX + 1.0 );
+	return min + ( v*(max-min) );
+}
```
```suggestion
/**
* Generates a random number on the interval [0,1).
*
* @return random number
*/
static float rand_float( void ) {
	int r = rand();
	return (float)r / ( (float)RAND_MAX + 1.0f );
}
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+*/
+static double benchmark( void ) {
+	double elapsed;
+	float x[ 100 ];
+	float y;
+	double t;
```
```suggestion
	float x[ 100 ];
	double t;
	float y;
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+	double t;
+	int i;
+
+    for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( -50.0, 50.0 );
+	}
```
```suggestion
	for ( i = 0; i < 100; i++ ) {
		x[ i ] = ( 100.0f * rand_float() ) - 50.0f;
	}

```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/docs/repl.txt:
```diff
@@ -0,0 +1,28 @@
+
+{{alias}}( x )
+    Evaluates the natural exponential function (single-precision).
```
@vivekmaurya001 This description must match with what we have in other files.

vivekmaurya001 replied:
done !

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/docs/repl.txt:
```diff
@@ -0,0 +1,28 @@
+    Examples
+    --------
+    > var y = {{alias}}( 4.0 )
+    54.598148345947266
```
```suggestion
    ~54.5982
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/docs/repl.txt:
```diff
@@ -0,0 +1,28 @@
+    > var y = {{alias}}( 4.0 )
+    54.598148345947266
+    > y = {{alias}}( -9.0 )
+    0.00012340980174485594
```
```suggestion
    ~1.234e-4
```

gunjjoshi commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

gunjjoshi committed:
Apply suggestions from code review

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented:
Left an initial review, some changes must be made before landing this PR.

gunjjoshi committed:
Update README.md

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update repl.txt

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/expf/examples/c/example.c:
```diff
@@ -0,0 +1,38 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+static float random_uniform( const float min, const float max ) {
+	float v = (float)rand() / ( (float)RAND_MAX + 1.0 );
+	return min + ( v*(max-min) );
+}
```
@vivekmaurya001 We don't need to make a custom function here. You can refer: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/fmodf/examples/c/example.c

kgryte replied:
Agree with @gunjjoshi. I will say that we have been creating custom functions in the `stats/base/dists/*` work, in order to avoid repeated RV generation logic. I am personally fine with either approach. In `stats`, it is more necessary due to more complex parameterization needs. Here, it is less so.

vivekmaurya001 replied:
i removed the custom function !

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 commented:
ohh , sorry i renamed this branch and it got closed i will do with required changes

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
feat: add math/base/special/expf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 committed:
Update README.md

Signed-off-by: Vivek maurya <155618190+vivekmaurya001@users.noreply.github.com>

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 commented:
@gunjjoshi , done all the final changes

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into Add-C-implementations-to-base-special-math-functions

vivekmaurya001 committed:
feat: add math/base/special/expf

vivekmaurya001 committed:
Update tsconfig.json

Signed-off-by: Vivek maurya <155618190+vivekmaurya001@users.noreply.github.com>

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Add-C-implementations-to-base-special-math-functions

anandkaranubc committed:
fix: use correct f32 emulation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: add hex values as comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: update runner file and generate correct fixtures

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: update tests according to current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
fix: correct C implementation and tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: update random value generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: update examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update metadata

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: update function description

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: update function return value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update copyright years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
Thanks @vivekmaurya001 for working on it. That will be a great and important addition! The PR looks ready to be merged. 🚀

kgryte commented:
@Planeshifter Mind giving this one a final lookover before merge?

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+
+/**
```
There's a leading blank line at the start of this file that should be removed. The license comment should begin on line 1.

```suggestion
/**
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+*/
+
+#include "stdlib/math/base/napi/unary.h"
+#include "stdlib/math/base/special/expf.h"
```
The include order should have the package's own header first, then stdlib dependencies. 

```suggestion
#include "stdlib/math/base/special/expf.h"
#include "stdlib/math/base/napi/unary.h"
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,214 @@
+	r = hi - lo;
+	t = r * r;
+	c = r - ( t * polyval_p( t ) );
+	y = 1.0f - ( lo - ( ( r*c )/( 2.0f-c ) ) - hi );
```
Let's add spaces around the operators to match the stdlib style guide and the reference implementation in `exp/src/main.c` (line 120).

```suggestion
	y = 1.0f - ( lo - ( ( r * c ) / ( 2.0f - c ) ) - hi );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,214 @@
+* -   The hexadecimal values included in the source code are the intended ones for the used constants. Decimal values may be used, provided that the compiler will convert from decimal to binary accurately enough to produce the intended hexadecimal values.
+*
+* @param x    input value
+* @return	  output value
```
The `@return` tag alignment should use spaces, not a tab character.

```suggestion
* @return     output value
```

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Add-C-implementations-to-base-special-math-functions

Planeshifter reviewed (changes requested):
Some final comments that should be addressed. Excited to land this PR soon!

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
style: remove leading empty line

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/lib/main.js:
```diff
@@ -0,0 +1,201 @@
+var INVLN2 = f32( 1.4426950216e+00 );     // 0x3FB8AA3B
+var OVERFLOW = f32( 8.8721679688e+01 );   // 0x42B17180
+var UNDERFLOW = f32( -1.0397208405e+02 ); // 0xC2CFF1B5
+var NEARZERO = f32( 1.0 / (1 << 14) );    // 0x39000000
+var NEG_NEARZERO = f32( -NEARZERO );      // 0xB9000000
```
The `NEARZERO` constant computes `1.0 / (1 << 14)` = `2^-14` = `0x38800000`, but the comment says `0x39000000` which is actually `2^-13`. The FreeBSD reference uses the hex value `0x39000000` as the threshold. This should be `1 << 13` to match the annotated hex and the reference implementation.

Same thing for `NEG_NEARZERO` on the next line — it inherits the wrong value.

```suggestion
var NEARZERO = f32( 1.0 / (1 << 13) );    // 0x39000000
var NEG_NEARZERO = f32( -NEARZERO );      // 0xB9000000
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,214 @@
+static const float INVLN2 = 1.4426950216e+00;         // 0x3FB8AA3B
+static const float EXP_OVERFLOW = 8.8721679688e+01;   // 0x42B17180
+static const float EXP_UNDERFLOW = -1.0397208405e+02; // 0xC2CFF1B5
+static const float NEARZERO = 1.0f / (1 << 14);       // 0x39000000
+static const float NEG_NEARZERO = -NEARZERO;          // 0xB9000000
```
Same `NEARZERO` mismatch as in the JS code — `1 << 14` gives `2^-14` (`0x38800000`), not `2^-13` (`0x39000000`) as annotated. Should be `1 << 13`.

```suggestion
static const float NEARZERO = 1.0f / (1 << 13);       // 0x39000000
static const float NEG_NEARZERO = -NEARZERO;          // 0xB9000000
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,214 @@
+#include "stdlib/constants/float32/pinf.h"
+#include <stdint.h>
+
+static const float LN2_HI = 6.9314575195e-01;         // 0x3F317200
+static const float LN2_LO = 1.4286067653e-06;         // 0x35BFBE8E
+static const float INVLN2 = 1.4426950216e+00;         // 0x3FB8AA3B
+static const float EXP_OVERFLOW = 8.8721679688e+01;   // 0x42B17180
+static const float EXP_UNDERFLOW = -1.0397208405e+02; // 0xC2CFF1B5
```
These `static const float` literals should have the `f` suffix to be consistent with other stdlib float packages (e.g., `lnf`, `acosf`).

```suggestion
static const float LN2_HI = 6.9314575195e-01f;         // 0x3F317200
static const float LN2_LO = 1.4286067653e-06f;         // 0x35BFBE8E
static const float INVLN2 = 1.4426950216e+00f;         // 0x3FB8AA3B
static const float EXP_OVERFLOW = 8.8721679688e+01f;   // 0x42B17180
static const float EXP_UNDERFLOW = -1.0397208405e+02f; // 0xC2CFF1B5
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/lib/main.js:
```diff
@@ -0,0 +1,201 @@
+*
+* ## Notice
+*
+* The following copyrights, licenses, and long comment were part of the original implementation available as part of{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}, The implementation follows the original, but has been modified for JavaScript.
```
The Notice block has a few formatting issues compared to the established pattern used in other FreeBSD-derived packages.
Per standard form, should be

```suggestion
* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}. The implementation follows the original, but has been modified for JavaScript.
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/lib/expmultif.js:
```diff
@@ -0,0 +1,76 @@
+*
+* ## Notice
+*
+* The following copyrights, licenses, and long comment were part of the original implementation available as part of{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}, The implementation follows the original, but has been modified for JavaScript.
```
```suggestion
* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}. The implementation follows the original, but has been modified for JavaScript.
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/src/main.c:
```diff
@@ -0,0 +1,214 @@
+*
+* ## Notice
+*
+* The following copyrights, licenses, and long comment were part of the original implementation available as part of{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}, The implementation follows the original, but has been modified according to project conventions.
```
Same Notice block formatting issues as in the JS files — missing `[FreeBSD]` link text, missing space before `{@link`, "copyrights, licenses" should be singular, and comma should be a period after the URL.

Planeshifter replied:
```suggestion
* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}. The implementation follows the original, but has been modified according to project conventions.
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/lib/main.js:
```diff
@@ -0,0 +1,201 @@
+	} else {
+		k = truncf( f32( f32( INVLN2*x ) + HALF ) );
+	}
+	hi = f32( x - f32( k*LN2_HI ) );
```
Missing spaces around binary operators here and on lines 190 and 192. Our style guide requires spaces around operators (e.g., `a + b`, `a * b`).

```suggestion
		k = truncf( f32( f32( INVLN2 * x ) - HALF ) );
	} else {
		k = truncf( f32( f32( INVLN2 * x ) + HALF ) );
	}
	hi = f32( x - f32( k * LN2_HI ) );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,63 @@
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = expf( x[ i%x.length ] );
```
Same spacing issue — `i%x.length` should be `i % x.length`.

```suggestion
		y = expf( x[ i % x.length ] );
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/package.json:
```diff
@@ -0,0 +1,149 @@
+{
+  "name": "@stdlib/math/base/special/expf",
+  "version": "0.0.0",
+  "description": "Evaluates the natural exponential function for a single-precision floating-point number.",
```
The description should use imperative mood: "Evaluate..." instead of "Evaluates..." (see `sqrtf` which uses "Compute...").

```suggestion
  "description": "Evaluate the natural exponential function for a single-precision floating-point number.",
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/include/stdlib/math/base/special/expf.h:
```diff
@@ -0,0 +1,37 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef STDLIB_MATH_BASE_SPECIAL_EXPF_H
+#define STDLIB_MATH_BASE_SPECIAL_EXPF_H
```
There should be a blank line between the closing `*/` of the license block and the `#ifndef` include guard.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/expf/docs/repl.txt:
```diff
@@ -0,0 +1,28 @@
+    NaN
+
+    See Also
+    --------
```
Missing a trailing blank line after the `See Also` dashes. Compare with `sqrtf/docs/repl.txt` which has an empty line after `--------`.

kgryte commented:
Once this PR is merged, we need to fix `math/base/special/coshf`: https://github.com/stdlib-js/stdlib/pull/10177

Planeshifter reviewed (changes requested):
@kgryte Looks like PR is not quite ready yet after all. Did another pass on it and left comments.

anandkaranubc committed:
Merge branch 'develop' into Add-C-implementations-to-base-special-math-functions

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: na
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
I think this should be ready to merge.

cc: @kgryte

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/powf/lib/polyval_l.js:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2025 The Stdlib Authors.
+* Copyright (c) 2026 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2025 The Stdlib Authors.
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/powf/lib/polyval_p.js:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2025 The Stdlib Authors.
+* Copyright (c) 2026 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2025 The Stdlib Authors.
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/powf/lib/polyval_w.js:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2025 The Stdlib Authors.
+* Copyright (c) 2026 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2025 The Stdlib Authors.
```

anandkaranubc committed:
Apply suggestions from code review

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/math/base/special/expf/lib/expmultif.js:
```diff
@@ -0,0 +1,76 @@
+*
+* ## Notice
+*
+* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_expf.c}. The implementation follows the original, but has been modified for JavaScript.
```
@anandkaranubc This is not the right attribution. See https://github.com/stdlib-js/stdlib/blob/75dab5bc88fd1d882d0e39cbbb63e0c5d81a9b32/lib/node_modules/%40stdlib/math/base/special/exp/lib/main.js#L19.

kgryte committed:
docs: fix license

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: fix license

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: fix license

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: fix license

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after clean-up and proper attribution.
